### PR TITLE
fix windows error #418

### PR DESCRIPTION
**Pull request by yhnu**
#412 #418

### CODE REVIEW

**Review by Nazeehe — APPROVED** [review]

File: lua/packer/plugin_utils.lua [hunk 1]
```diff
@@ -163,7 +163,7 @@ plugin_utils.load_plugin = function(plugin)
           error(files)
         end
       elseif #files > 0 then
-        for _, file in ipairs(files) do vim.cmd('silent exe "source ' .. file .. '"') end
+        for _, file in ipairs(files) do file = file:gsub("\\", "/") vim.cmd('silent exe "source ' .. file .. '"') end
```
Comment (wbthomason) [inline lua/packer/plugin_utils.lua:166]: Nitpick: if this `for` loop is no longer a single statement, then it should be split to multiple lines.

Also: I'm surprised that replacing with `/` works on Windows - is there somewhere in the Vim/Neovim help that indicates why this is OK?

**Review by wbthomason — CHANGES_REQUESTED** [review]
Thanks, Windows errors are very difficult for me to find since I have no Windows system for testing! There's one style tweak I'd like to make, but otherwise this is probably good to go (though I admit I'm surprised that using forward slashes works as expected, and I'd like to document why this is needed).

**Comment by Nazeehe** [comment]
> Thanks, Windows errors are very difficult for me to find since I have no Windows system for testing! There's one style tweak I'd like to make, but otherwise this is probably good to go (though I admit I'm surprised that using forward slashes works as expected, and I'd like to document why this is needed).

Yeah, I run on Windows so I can certainly help bring bugs back here. Forward slashes are acceptable for Windows APIs, that's why they work.

**Comment by wbthomason** [comment]
All Neovim Windows APIs?

I'm ok to merge this once we fix the formatting nitpick I commented on in my review. Thanks again!

**Comment by bugabinga** [comment]
> All Neovim Windows APIs?

Yes, but more specifically all win32-APIs.
On Windows `\` is interchangeable with `/` in paths.

https://stackoverflow.com/questions/38428561/difference-between-forward-slash-and-backslash-in-file-path?rq=1

I would not be surprised if the whole path separator handling in https://github.com/wbthomason/packer.nvim/blob/master/lua/packer/util.lua#L40 can simply be deleted.

Simply use forward slashes everywhere.

**Comment by akinsho** [comment]
@wbthomason I wonder if regarding the comment above we could test out a PR that removes the path handling stuff for windows, get some windows users to test and see if it breaks anything. I guess though that there might be edge cases 🤔 or maybe packer compiled files that currently contain a backlash that might break(although if they are interchangeable they shouldn't)

**Comment by wbthomason** [comment]
Thanks, this is good to know! I thought forward slashes only worked in limited contexts on Windows (e.g. in Powershell, but not in general).

Given that, I like @akinsho's idea to try simplifying the code. In the meanwhile, I'll apply the formatting change I still wanted here and merge this PR.

File: lua/packer/plugin_utils.lua (hunk 1 above)
Comment (wbthomason) [reply lua/packer/plugin_utils.lua]: Applied the reformat myself in 53b7e1d.

**Review by wbthomason — APPROVED** [review]
Seems good to go, thanks!
